enigma/src/main/java/cuchaz/enigma/analysis/StructureTreeOptions.java

```diff
@@ -0,0 +1,60 @@
+package cuchaz.enigma.analysis;
+
+import cuchaz.enigma.utils.I18n;
+
+public record StructureTreeOptions(
+        ObfuscationVisibility obfuscationVisibility,
+        DocumentationVisibility documentationVisibility,
+        SortingOrder sortingOrder) {
+
+    public enum SortingOrder {
+        DEFAULT("default"),
+        A_Z("a_z"),
+        Z_A("z_a");
+
+        private final String key;
+
+        SortingOrder(String key) {
+            this.key = key;
+        }
+
+        @Override
+        public String toString() {
+            return I18n.translate("structure.options.sorting." + this.key);
```

Comment:
So `toString` is the only way the j combo box displays text? Guess we will use it then...

Reply:
It depends on the renderer ([`ListCellRenderer`](https://docs.oracle.com/en/java/javase/16/docs/api/java.desktop/javax/swing/ListCellRenderer.html)) you use. The defaults (`BasicComboBoxRenderer` in Metal, `DefaultListCellRenderer` in FlatLaf) happen to use `toString()`.

Reply:
No, it's not, there's an alternative, that's what I was talking about in the other comment:

> Especially because I think the alternative is considerably more annoying to implement.

Let me actually look at it to see what would need to be done for that because I was going from memory there.

Reply:
Looks like you just need a custom ListCellRenderer that you set on the combobox with setCellRenderer. A custom ComboBoxEditor isn't needed since these comboboxes aren't editable. This actually shouldn't be too bad since there already is an AbstractListCellRenderer class that's used in the search dialog that should make this more bearable. @YanisBft could you implement this?

Reply:
If the list cell renderers supplied implement `ListCellRenderer<Object>`, we can just make a delegate one that converts the object to a desired string and then proceed to call the delegate old renderer.

Reply:
Is it really needed? I mean, I still think these combo boxes should be replaced by fancy buttons with icons, because it's kind of ugly right now.

Reply:
Sure, we can replace the list cell renderer to just render fancy buttons with icons in a future commit.

Reply:
Nevermind it was simple, done